### ISSUE TITLE
Update module-accounts close_account to remove proxies as well

### ISSUE THREAD

**Issue by xlc** [AcalaNetwork/Acala#549]
It was not possible to call `close_account` for proxy created via module-nft because `close_account` requires no other reserved token and proxy is the only way to call `close_account` which means it will have reserved token.

The solution is to make `close_account` remove all the proxies first and then attempt to close the account.

Before https://github.com/paritytech/substrate/issues/7557 is ready, we can just do this to remove proxies

```
let (_, proxy_deposit) = pallet_proxy::Proxies::<T>::take(&class_info.owner);
<T as pallet_proxy::Trait>::Currency::unreserve(&class_info.owner, proxy_deposit);
```

**Comment by xlc** [comment]
No longer needed after #566
